subprojects/core/src/test/groovy/org/gradle/api/internal/attributes/DefaultMutableAttributeContainerTest.groovy

```diff
@@ -359,4 +336,19 @@ class DefaultMutableAttributeContainerTest extends Specification {
             container.getAttribute(attribute)
         }
     }
+
+    def "cannot define two attributes with the same name but different types"() {
```

Comment:
You cannot define two attributes on an immutable attribute container, since you cannot define _any_ attributes on an immutable attribute container.

This was just copied from `DefaultConfigurationSpec`, where it was testing the behavior of the mutable attribute container. I just moved it to a better place